ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt

```diff
@@ -31,13 +33,14 @@ internal class JsClientEngine(override val config: HttpClientEngineConfig) : Htt
     @OptIn(InternalAPI::class)
     override suspend fun execute(data: HttpRequestData): HttpResponseData {
         val callContext = callContext()
+        val clientConfig = data.attributes[CLIENT_CONFIG]
```

Comment:
why pass it through data? there is a property `config` in class

Reply:
cause we need client config, not engine

Reply:
Ok. Let's create a ticket to pass client config in client factory `HttpClientEngineFactory.create` method in 3.0.0 for a proper fix.